### DISCUSSION TITLE
Does anyone know of a similar module for MCP2200

### DISCUSSION THREAD

**Discussion by RufusVS** [electronicayciencia/EasyMCP2221#6] [Q&A] [answered]
I came across this library because the MCP2221 is a cousin chip to the MCP2200.  I have a Denkovi relay board that uses the MCP2200 chip that I am trying to speak the HID interface which controls the relays, and am having a hard time with it as far as connecting to the HID usb port.  I keep getting closer but if I reinstall It seems to get a different driver.  I finally found the MCP2221driver installer at Microchip and it seems to be what I want, and this library can actually get data off the chip at start if I change the buffer size to16 instead of 64.  I am going to try to make a minimal port of this and will report back but it would be a lot easier if someone has already done it.

I am doing this in Windows 11

Update: I did not find a similar module for mcp2200, and if I get ambitious, I might write one, but after pulling together documentation from Microchip, mcp2200 code in different language and for different OS (Linux).  And looking at this code and other code for other Microchip hid chips, I was able to boil it down along will installing and deleting different drivers, the code for toggling a relay. (Note: I need to create a "configure" code as well, in case GPIO isn't set the correct direction)  But with configured hardware, here's the python turnon.py code:

```
import hid
import sys

MCP2200_SET_CLEAR_OUTPUTS=0x08
RELAY_NUM = 1
RELAY_MASK = 0x0F

if len(sys.argv) > 1:
    RELAY_NUM = int(sys.argv[1])

SET_MASK = 1 << (RELAY_NUM - 1)
CLR_MASK = RELAY_MASK ^ SET_MASK

h = hid.device()
h.open(0x04d8, 0x00df)                  # Microchip MCP2200
buf = ([0] +                            # report 0
    [MCP2200_SET_CLEAR_OUTPUTS] +       # relay set/clear
    [0]*10 +                            # unused
    [SET_MASK] + 
    [CLR_MASK] +    
    [0]*3)
t = h.write(buf)
h.close()
```

**Comment by electronicayciencia** [comment]
Hello. I don't have any MCP2200 to test. But tell me if I can help you with anything.

**Reply by RufusVS** [reply]
I updated the original comment.  I distilled all my information down to workable code.  I may possibly create an EasyMCP2200 module if I have time, but work is pretty busy right now.  It would be far simpler than EasyMCP2221 because there are far fewer functions in the chip.

**Comment by RufusVS** [comment, marked as answer]
Actual answer in original question which I edited.
